lib/server.go

```diff
@@ -1144,6 +1146,8 @@ func (srv *Server) _handleSnapshot(pp *Peer, msg *MsgDeSoSnapshotData) {
 		"<%v>, Last entry: <%v>), (number of entries: %v), metadata (%v), and isEmpty (%v), from Peer %v",
 		msg.SnapshotChunk[0].Key, msg.SnapshotChunk[len(msg.SnapshotChunk)-1].Key, len(msg.SnapshotChunk),
 		msg.SnapshotMetadata, msg.SnapshotChunk[0].IsEmpty(), pp)))
+	// Free up a slot in the operationQueueSemaphore, now that a chunk has been processed.
+	srv.snapshot.FreeOperationQueueSemaphore()
```

Comment:
Is this the correct place to do this? Should it happen towards the end of the function? Would it make sense to defer this?

Reply:
Yeah, it's fine to free the chunk queue as soon as we receive a more-or-less valid chunk. The logic below this point checks for the chunk's validity; if we find it invalid, we generally disconnect the peer or restart the node. If the chunk is valid, we process it, which is expensive. Freeing the channel here allows us to get the next chunk a little bit faster compared to if we postponed this operation to the end of the function.